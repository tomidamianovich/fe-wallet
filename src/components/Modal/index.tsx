import React, { ReactElement } from 'react';
import './styles/index.scss';

type ButtonProps = {
  label: string,
  ariaLabel: string,
  handler: () => void,
  backgroundColor?: string,
  color?: string,
  disabled?: boolean
};

type Props = {
  header: string,
  content: ReactElement<{}>,
  actions?: ButtonProps[],
  handleModalVisibility: () => void
};

const Modal: React.FC<Props> = ({
  header,
  content,
  actions,
  handleModalVisibility
}) => {
  return(
    <div className="fe-wallet__modal">
      <div className="fe-wallet__modal__content">
        <div className="fe-wallet__modal__content__header">
          <h2>{header}</h2>
          <span
            className="fe-wallet__modal__content__header__close" 
            onClick={handleModalVisibility}>
              &times;
          </span>
        </div>
        <div className="fe-wallet__modal__content__body">
          <div className="fe-wallet__modal__content__body__content">
            {content}
          </div>
          <div className="fe-wallet__modal__content__body__buttons">
            { actions?.map( action => 
              <button
                aria-label={action.ariaLabel} 
                onClick={action.handler}
                disabled={action.disabled}
                style={{
                  backgroundColor: action.backgroundColor,
                  color: action.color
                }}>
                {action.label}
              </button>  
            )}
          </div>
        </div>
      </div>
    </div>
  )
};

export default Modal;