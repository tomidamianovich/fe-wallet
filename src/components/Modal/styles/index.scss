
@import "../../../containers/App/styles/colours.scss";

/* Modal Content */
.fe-wallet__modal {
  position: fixed; 
  z-index: 1; 
  padding-top: 100px;
  left: 0;
  top: 0;
  width: 100%; 
  height: 100%;
  overflow: auto;
  background-color: $tertiary-background-color;

  &__content {
    position: relative;
    background-color: #fefefe;
    margin: auto;
    padding: 0;
    border: 1px solid $primary-border-color;
    width: 50%;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2),0 6px 20px 0 rgba(0,0,0,0.19);
    animation-name: animatetop;
    animation-duration: 0.4s;

    @media screen and (max-width: 500px) {
      width: 90%;
      margin: 0 5%;
    }
    
    &__header {
      padding: 2px 16px;
      background-color: $primary-background-color;
      display: flex;
      justify-content: space-between;
      align-items: center;
      color: $primary-text-color;
      box-shadow: 0px 2px 3px 0px $primary-border-color;
      &__close {
        cursor: pointer;
        font-size: 32px;
      }
    }
    
    &__body {
      padding: 15px;
      display: flex;
      align-items: stretch;
      flex-direction: column;
      flex-wrap: wrap;
      justify-content: space-around;
      &__buttons {
        padding: .5rem 0 0 0;
        button {
          padding: .5rem 1rem;
          margin: .25rem;
          border: none;
          box-shadow: 0px 0px 0px 0px $tertiary-background-color;
          border-radius: 3px;
          &:hover {
            background-color: $tertiary-background-color;
            cursor: pointer;
          }
        }
      }
    }
  }
  @keyframes animatetop {
    from {top: -300px; opacity: 0}
    to {top: 0; opacity: 1}
  }
}
  