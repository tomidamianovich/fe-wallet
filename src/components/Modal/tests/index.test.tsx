import { render } from '@testing-library/react';
import Modal from '../index';

const mockFn = () => {}

const buttons = [{
  label:"Test",
  ariaLabel:"test",
  handler: mockFn,
  backgroundColor:"test",
  color:"test"
},
{
  label:"Test 2",
  ariaLabel:"tes 2t",
  handler: mockFn,
  backgroundColor:"test 2",
  color:"test 2"
}]

test('renders Modal with 1 button', () => {
  
  const { asFragment } = render(
    <Modal 
      handleCloseButton={mockFn} 
      header="Header"
      content={ <div>Test</div> }
      actions={[buttons[0]]}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders Modal with 2 buttons', () => {
  
  const { asFragment } = render(
    <Modal 
      handleCloseButton={mockFn} 
      header="Header"
      content={ <div>Test</div> }
      actions={buttons}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders Modal without buttons', () => {
  
  const { asFragment } = render(
    <Modal 
      handleCloseButton={mockFn} 
      header="Header"
      content={ <div>Test</div> }
    />
  );
  expect(asFragment()).toMatchSnapshot();
});