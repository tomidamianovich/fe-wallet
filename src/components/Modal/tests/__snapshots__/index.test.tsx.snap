// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Modal with 1 button 1`] = `
<DocumentFragment>
  <div
    class="fe-wallet__modal"
  >
    <div
      class="fe-wallet__modal__content"
    >
      <div
        class="fe-wallet__modal__content__header"
      >
        <h2>
          Header
        </h2>
        <span
          class="fe-wallet__modal__content__header__close"
        >
          ×
        </span>
      </div>
      <div
        class="fe-wallet__modal__content__body"
      >
        <div
          class="fe-wallet__modal__content__body__content"
        >
          <div>
            Test
          </div>
        </div>
        <div
          class="fe-wallet__modal__content__body__buttons"
        >
          <button
            aria-label="test"
          >
            Test
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Modal with 2 buttons 1`] = `
<DocumentFragment>
  <div
    class="fe-wallet__modal"
  >
    <div
      class="fe-wallet__modal__content"
    >
      <div
        class="fe-wallet__modal__content__header"
      >
        <h2>
          Header
        </h2>
        <span
          class="fe-wallet__modal__content__header__close"
        >
          ×
        </span>
      </div>
      <div
        class="fe-wallet__modal__content__body"
      >
        <div
          class="fe-wallet__modal__content__body__content"
        >
          <div>
            Test
          </div>
        </div>
        <div
          class="fe-wallet__modal__content__body__buttons"
        >
          <button
            aria-label="test"
          >
            Test
          </button>
          <button
            aria-label="tes 2t"
          >
            Test 2
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Modal without buttons 1`] = `
<DocumentFragment>
  <div
    class="fe-wallet__modal"
  >
    <div
      class="fe-wallet__modal__content"
    >
      <div
        class="fe-wallet__modal__content__header"
      >
        <h2>
          Header
        </h2>
        <span
          class="fe-wallet__modal__content__header__close"
        >
          ×
        </span>
      </div>
      <div
        class="fe-wallet__modal__content__body"
      >
        <div
          class="fe-wallet__modal__content__body__content"
        >
          <div>
            Test
          </div>
        </div>
        <div
          class="fe-wallet__modal__content__body__buttons"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
